tests/test_compiled_engine.py

```
@@ -465,12 +468,208 @@ def test_compiled_jaxpr_has_no_host_callbacks(small_sim_params):
+    monitor_type_names = {cls.__name__ for cls in COMPILED_DEVICE_REGISTRY.monitor_types}
+
+    assert source_type_names == {"GaussianSource", "ModeSource"}
+    assert monitor_type_names == {"Monitor"}
+
+
+def test_run_compiled_rejects_unsupported_device_type(small_sim_params):
+    class DummyLegacySource:
+        def inject(self, fields, t, dt, current_step, resolution, design):
+            del fields, t, dt, current_step, resolution, design
+
+    wl, dx, _dt, domain, _steps, t, _signal = small_sim_params
```

Ruff (0.15.7)

[warning] Unpacked variable `wl` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)